pkg/handler/wallet/wallet.go

```diff
@@ -118,30 +118,45 @@ func (h *Handler) GetOne(c *gin.Context) {
 // Track     	godoc
 // @Summary     Track new wallet
 // @Description Track new wallet
-// @Tags        Wallet
+// @Tags        WatchList
 // @Accept      json
 // @Produce     json
 // @Param       id   			path  string true  "user ID"
 // @Param       request  	body 	request.TrackWalletRequest true "req"
 // @Success     200 {object} 		response.ResponseMessage
-// @Router      /users/:id/wallets [post]
+// @Router      /users/{id}/watch-lists/wallets/track [post]
 func (h *Handler) Track(c *gin.Context) {
 	var req request.TrackWalletRequest
 	if err := c.BindJSON(&req); err != nil {
 		h.log.Error(err, "[handler.Wallet.Track] BindJSON() failed")
 		c.JSON(http.StatusBadRequest, response.CreateResponse[any](nil, nil, err, nil))
 		return
 	}
-	err := h.entities.TrackWallet(req)
+
+	userID := c.Param("id")
+	if userID == "" {
+		err := errors.New("user ID is required")
+		h.log.Error(err, "[handler.Wallet.Track] not enough params")
+		c.JSON(http.StatusBadRequest, response.CreateResponse[any](nil, nil, err, nil))
+		return
+	}
+
+	req.UserID = userID
+
+	mod, err := req.RequestToUserWalletWatchlistItemModel()
```

Comment:
1. why dont we use tag `enums` for `chain_type` and `type` to validate
2. why we need to map `TrackWalletRequest` to `model.UserWalletWatchlistItem`

Reply:
1. Same, just want to identify the reason why the variable hold specific value. It mean because it is a ChainType, so it can be `eth`, `sui`...ble ble. 
2. Because I want to validation is done on the handler.